volatility3/framework/plugins/linux/capabilities.py

```diff
@@ -0,0 +1,228 @@
+            ),
+            requirements.PluginRequirement(
+                name="pslist", plugin=pslist.PsList, version=(2, 0, 0)
+            ),
+            requirements.ListRequirement(
+                name="pids",
+                description="Filter on specific process IDs.",
+                element_type=int,
+                optional=True,
+            ),
+            requirements.BooleanRequirement(
+                name="inheritable",
+                description="Show only inheritable capabilities in human-readable strings.",
+                optional=True,
+            ),
+            requirements.BooleanRequirement(
+                name="permitted",
+                description="Show only permitted capabilities in human-readable strings.",
+                optional=True,
+            ),
+            requirements.BooleanRequirement(
+                name="effective",
+                description="Show only effective capabilities in human-readable strings.",
+                optional=True,
+            ),
+            requirements.BooleanRequirement(
+                name="bounding",
+                description="Show only bounding capabilities in human-readable strings.",
+                optional=True,
+            ),
+            requirements.BooleanRequirement(
+                name="ambient",
+                description="Show only ambient capabilities in human-readable strings.",
+                optional=True,
+            ),
+        ]
+
+    def _check_capabilities_support(self):
+        """Checks that the framework supports at least as much capabilities as
+        the kernel being analysed. Otherwise, it shows a warning for the
+        developers.
+        """
+        vmlinux = self.context.modules[self.config["kernel"]]
+
+        kernel_cap_last_cap = vmlinux.object(
+            object_type="int", offset=kernel_cap_last_cap
+        )
+        vol2_last_cap = extensions.kernel_cap_struct.get_last_cap_value()
+        if kernel_cap_last_cap > vol2_last_cap:
+            vollog.warning(
+                "Developers: The supported Linux capabilities of this plugin are outdated for this kernel"
+            )
+
+    @staticmethod
+    def _decode_cap(cap: interfaces.objects.ObjectInterface) -> str:
+        """Returns a textual representation of the capability set.
+        The format is a comma-separated list of capabilitites. In order to
+        summarize the output and if all the capabilities are enabled, instead of
+        the individual capabilities, the special name "all" will be shown.
+
+        Args:
+            cap: Kernel capability object. Usually a 'kernel_cap_struct' struct
+
+        Returns:
+            str: A string with a comma separated list of decoded capabilities
+        """
+        if isinstance(cap, renderers.NotAvailableValue):
+            return cap
+
+        cap_value = cap.get_capabilities()
+        if cap_value == 0:
+            return "-"
+
+        CAP_FULL = 0xFFFFFFFF
```

Comment:
Could we move CAP_FULL to the constants file please?  Easier to keep things like this all in one place, even if it's incredibly unlikely to change...  5:)